### PR TITLE
helm: use correct nodeSelector for k8s 1.25+

### PR DESCRIPTION
**Pull request by jackfrancis**
**What type of PR is this?**

/kind failing-test
/kind api-change

**What this PR does / why we need it**:

This PR sets the correct nodeSelector in the helm chart so that it works with k8s 1.25+

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: charts/latest/azuredisk-csi-driver/templates/csi-azuredisk-controller.yaml [hunk 1]
```diff
@@ -26,7 +26,11 @@ spec:
       nodeSelector:
         kubernetes.io/os: linux
         {{- if .Values.controller.runOnMaster}}
```
Comment (marosset) [inline charts/latest/azuredisk-csi-driver/templates/csi-azuredisk-controller.yaml:28]: Should we add a new `runOnControlPlane` setting and have this line check if either is set and the apply the correct node-selector?

That is more work now but probably better in the long run.

Comment (marosset) [reply charts/latest/azuredisk-csi-driver/templates/csi-azuredisk-controller.yaml:28]: fair enough

Comment (andyzhangx) [reply charts/latest/azuredisk-csi-driver/templates/csi-azuredisk-controller.yaml:28]: I think we could remove `controller.runOnMaster` chart, replace with `controller.runOnControlPlane` directly

**Comment by andyzhangx** [comment]
this PR should have fixed the test failure on 1.25+: https://github.com/kubernetes-sigs/azuredisk-csi-driver/pull/1362

File: charts/latest/azuredisk-csi-driver/templates/csi-azuredisk-controller.yaml (hunk 1 above)
Comment (jackfrancis) [reply charts/latest/azuredisk-csi-driver/templates/csi-azuredisk-controller.yaml:28]: Won't this break back-compat for existing users? (or has a new version of the chart w/ this configuration never been released?)

Comment (andyzhangx) [reply charts/latest/azuredisk-csi-driver/templates/csi-azuredisk-controller.yaml:28]: that's ok for chart config of new versions, add a new `controller.runOnControlPlane` could be safer, and after a few versions, we could remove `controller.runOnMaste`

Comment (jackfrancis) [reply charts/latest/azuredisk-csi-driver/templates/csi-azuredisk-controller.yaml:28]: how about we do that as a follow-up change? in the meantime looks like we can merge this to fix broken tests

Comment (andyzhangx) [reply charts/latest/azuredisk-csi-driver/templates/csi-azuredisk-controller.yaml:28]: the windows test failure should already be fixed by https://github.com/kubernetes-sigs/azuredisk-csi-driver/pull/1362, no worry, I will provide a fix using `controller.runOnControlPlane` later

Comment (andyzhangx) [reply charts/latest/azuredisk-csi-driver/templates/csi-azuredisk-controller.yaml:28]: PR(https://github.com/kubernetes-sigs/azuredisk-csi-driver/pull/1366) would address the issue.

**Pull request closed without merging**